infrastructure/restapi/src/test/java/tech/pegasys/teku/infrastructure/restapi/endpoints/EndpointMetadataTest.java

```diff
@@ -330,7 +331,7 @@ void getRequestBody_shouldErrorIfSelectorDeterminesUndocumentedResult() {
 
     assertThatThrownBy(
             () -> metadata.getRequestBody(toStream("{\"value1\":\"FOO\"}"), Optional.empty()))
-        .isInstanceOf(IllegalStateException.class);
+        .isInstanceOf(MismatchedInputException.class);
```

Comment:
I think this is because we've removed the IllegalStateException, i just wonder if there's other places we've broken because of that, not sure what the coverage of tests is like - it does make me nervous changing outputs for things like this that are low level...
Eg. IllegalStateException would have been caught by the http handler, and we'll want to make sure this new exception is now correctly handled...

Reply:
Reverted it. Made a slight change refactoring `isEquivalentToDeserializableType` though